linalg/hypre.hpp

```diff
@@ -1003,9 +1063,94 @@ class HypreBoomerAMG : public HypreSolver
        construct A. */
    void SetElasticityOptions(ParFiniteElementSpace *fespace);
 
+#if MFEM_HYPRE_VERSION >= 21800
+   /** Hypre parameters to use AIR AMG solve for advection-dominated problems.
+   	   See "Nonsymmetric Algebraic Multigrid Based on Local Approximate Ideal
+   	   Restriction (AIR)," Manteuffel, Ruge, Southworth, SISC (2018),
+   	   DOI:/10.1137/17M1144350. Options: "distanceR" -> distance of neighbor
+   	   DOFs to buld restriction operator; options include 1, 2, and 15 (1.5).
+   	   Strings "prerelax" and "postrelax" indicate points to relax on:
+   	   F = F-points, C = C-points, A = all points. E.g., FFC -> relax on
+   	   F-points, relax again on F-points, then relax on C-points. */
+   void SetAdvectiveOptions(int distance=15,  const std::string &prerelax="",
+	                         const std::string &postrelax="FFC");
+
+   /// Expert option - consult hypre documentation/team
+   void SetStrongThresholdR(double strengthR)
+   { HYPRE_BoomerAMGSetStrongThresholdR(amg_precond, strengthR); }
```

Comment:
What did we decide about these "expert options"? I think I prefer to allow the user to call the hypre functions directly, and use the implicit casting.

Reply:
I kept them in the MFEM class so that the user (or me for that matter) doesn't have to pull up the hypre manual and find the functions every time they want to tune AMG. Much easier to just look in the AMG class here.

Reply:
Sounds good 👍